webapp/tests/model/test_eligibility_data.py

```diff
@@ -0,0 +1,1688 @@
+        try:
+            MarriedEligibilityData.parse_obj(valid_data)
+        except ValidationError:
+            self.fail("FamilienStandEligibilityData.parse_obj should not raise validation error")
+
+
+class TestWidowedEligibilityData(unittest.TestCase):
+
+    def test_if_marital_status_not_widowed_raise_validation_error(self):
+        invalid_marital_statuses = ['single', 'married', 'divorced', 'INVALID']
+        for invalid_marital_status in invalid_marital_statuses:
+            non_valid_data = {'marital_status': invalid_marital_status}
+            self.assertRaises(ValidationError, WidowedEligibilityData.parse_obj, non_valid_data)
+
+    def test_if_marital_status_widowed_then_raise_no_validation_error(self):
+        valid_data = {'marital_status': 'widowed'}
+        try:
+            WidowedEligibilityData.parse_obj(valid_data)
+        except ValidationError:
+            self.fail("FamilienStandEligibilityData.parse_obj should not raise validation error")
+
+
+class TestSingleEligibilityData(unittest.TestCase):
+
+    def test_if_marital_status_not_married_raise_validation_error(self):
+        invalid_marital_statuses = ['married', 'widowed', 'divorced', 'INVALID']
+        for invalid_marital_status in invalid_marital_statuses:
+            non_valid_data = {'marital_status': invalid_marital_status}
+            self.assertRaises(ValidationError, SingleEligibilityData.parse_obj, non_valid_data)
+
+    def test_if_marital_status_single_then_raise_no_validation_error(self):
+        valid_data = {'marital_status': 'single'}
+        try:
+            SingleEligibilityData.parse_obj(valid_data)
+        except ValidationError:
+            self.fail("FamilienStandEligibilityData.parse_obj should not raise validation error")
+
+
+class TestDivorcedEligibilityData(unittest.TestCase):
+
+    def test_if_marital_status_not_married_raise_validation_error(self):
+        invalid_marital_statuses = ['married', 'widowed', 'single', 'INVALID']
+        for invalid_marital_status in invalid_marital_statuses:
+            non_valid_data = {'marital_status': invalid_marital_status}
+            self.assertRaises(ValidationError, DivorcedEligibilityData.parse_obj, non_valid_data)
+
+    def test_if_marital_status_divorced_then_raise_no_validation_error(self):
+        valid_data = {'marital_status': 'divorced'}
+        try:
+            DivorcedEligibilityData.parse_obj(valid_data)
+        except ValidationError:
+            self.fail("FamilienStandEligibilityData.parse_obj should not raise validation error")
+
+
+class TestLongSeparateLivingEligibilityData(unittest.TestCase):
+
+    def test_if_marital_status_valid_and_separated_since_last_year_no_then_raise_validation_error(self):
+        non_valid_data = {'separated_since_last_year': 'no'}
+        with patch('app.model.eligibility_data.MarriedEligibilityData.parse_obj'):
+            self.assertRaises(ValidationError, LongSeparateLivingEligibilityData.parse_obj, non_valid_data)
```

Comment:
We talked about it and I got the logic behind the DecisionSteps. 👍

Reply:
I'll add more documentation for that. The reason is that we use the data types also to differentiate which branch we are taking (if the data is valid stay on the main branch and if it is not valid take the alternative branch). Unfortunately, this is not the most intuitive concept. However, it is quite useful because it makes using the data types for the steps easier.

I'll also rename the `LongSeparateLivingEligibilityData` to `SeparatedEligibilityData` and `ShortSeparateLivingEligibilityData` to `NotSeparatedEligibilityData`